sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -225,35 +208,32 @@ object GpuParquetScan {
     //     were written in that timezone and convert them to UTC timestamps.
     // Essentially this should boil down to a vector subtract of the scalar delta
     // between the configured timezone's delta from UTC on the timestamp data.
+    val schemaHasTimestamps = readSchema.exists { field =>
+      TrampolineUtil.dataTypeExistsRecursively(field.dataType, _.isInstanceOf[TimestampType])
+    }
     if (schemaHasTimestamps && sparkSession.sessionState.conf.isParquetINT96TimestampConversion) {
       meta.willNotWorkOnGpu("GpuParquetScan does not support int96 timestamp conversion")
     }
 
+    val schemaHasDates = readSchema.exists { field =>
+      TrampolineUtil.dataTypeExistsRecursively(field.dataType, _.isInstanceOf[DateType])
+    }
+
     sqlConf.get(SparkShimImpl.int96ParquetRebaseReadKey) match {
-      case "EXCEPTION" => if (schemaMightNeedNestedRebase) {
-        meta.willNotWorkOnGpu("Nested timestamp and date values are not supported when " +
-            s"${SparkShimImpl.int96ParquetRebaseReadKey} is EXCEPTION")
-      }
-      case "CORRECTED" => // Good
+      case "EXCEPTION" | "CORRECTED" => // Good
```

Comment:
Why is `EXCEPTION` ok now? Do we have any tests that exercise this path?

Reply:
`EXCEPTION` is handled somewhere else, after loading data. This is the planning step, thus we are OK to run on the GPU with it.

Reply:
We have test for it too. That is one of the two that I modified.

Reply:
I was confused because I didn't see `EXCEPTION` explicitly in the tests, but I checked the docs, and it is the default setting for `datetimeRebaseModeInRead`, so it is covered.